### PR TITLE
[5.4] Add Oracle 12c ansi pagination support

### PR DESCRIPTION
**Pull request by peyobr**
Tested on my environment. Thank you for watching.

### CODE REVIEW

File: src/Oci8/Query/Grammars/OracleGrammar.php [hunk 1]
```diff
@@ -83,6 +83,15 @@ protected function isPaginationable(Builder $query, array $components)
      */
     protected function compileAnsiOffset(Builder $query, $components)
     {
+        // Improved response time with FIRST_ROWS(n) hint for ORDER BY queries
+        if ($query->getConnection()->getConfig("server_version") == "12c") {
+            $components["columns"] = str_replace("select", "select /*+ FIRST_ROWS({$query->limit}) */", $components["columns"]);
+            $offset = $query->offset ?: 0;
+            $limit = $query->limit;
+            $components["limit"] = "offset $offset rows fetch next $limit rows only";
```
Comment (yajra) [inline src/Oci8/Query/Grammars/OracleGrammar.php]: Please add a new line before return.

**Comment by peyobr** [comment]
New line added.

At the moment don't see how to change Oci8QueryBuilderTest in order to work as expected...

**Comment by yajra** [comment]
Have tested the PR and seems to work well 👍 . 

Will try to fix the test and will merge when done. Thanks!

**Comment by peyobr** [comment]
OK Arjay! Thank you (y)

**Comment by robsontenorio** [comment]
@yajra @peyobr works like a charm. wont be merged?

**Comment by yajra** [comment]
Sorry for late response. I think we need to fix the tests fix for this to be merged. Thanks!

**Comment by yajra** [comment]
Was able to fix the tests. Will include this on the next release. Thanks!

**Pull request merged**